Octokit/Clients/CheckRunsClient.cs

```diff
@@ -0,0 +1,133 @@
+        {
+            Ensure.ArgumentNotNullOrEmptyString(reference, nameof(reference));
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+
+            return List(repositoryId, reference, checkRunRequest, ApiOptions.None);
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(string owner, string name, string reference, CheckRunRequest checkRunRequest)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNullOrEmptyString(reference, nameof(reference));
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+
+            return List(owner, name, reference, checkRunRequest, ApiOptions.None);
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(long repositoryId, long checkSuiteId, CheckRunRequest checkRunRequest)
+        {
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+            
+            return List(repositoryId, checkSuiteId, checkRunRequest, ApiOptions.None);
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(string owner, string name, long checkSuiteId, CheckRunRequest checkRunRequest)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+
+            return List(owner, name, checkSuiteId, checkRunRequest, ApiOptions.None);
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(long repositoryId, string reference, CheckRunRequest checkRunRequest, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(reference, nameof(reference));
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            throw new NotImplementedException();
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(string owner, string name, string reference, CheckRunRequest checkRunRequest, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNullOrEmptyString(reference, nameof(reference));
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            throw new NotImplementedException();
+        }
+
+        public Task<IReadOnlyList<CheckRun>> List(long repositoryId, long checkSuiteId, CheckRunRequest checkRunRequest, ApiOptions options)
+        {
+            Ensure.ArgumentNotNull(checkRunRequest, nameof(checkRunRequest));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            throw new NotImplementedException();
```

Comment:
Listing check runs for a reference doesn't follow the standard pagination rules for most other APIs. Advice?

Reply:
If it doesn't follow the conventions, you can exclude it here:

https://github.com/octokit/octokit.net/blob/3345f76fc9a499aca6f49defb345a7926ceb4c87/Octokit.Tests.Conventions/PaginationTests.cs#L96-L105

Reply:
why do you say it doesnt follow the standard rules?  From the response in the docs it indicates it should support "normal" pagination:

![image](https://user-images.githubusercontent.com/5425163/39755159-1d180b98-5308-11e8-9425-5a81c9921a5f.png)

Reply:
that said, if there is a case where it doesnt follow normal conventions, another option rather than excluding the whole client interface as @shiftkey mentioned, is that we also have an attributes which can suppress pagination conventions for naming and/or ApiOptions overloads, on a per method basis :+1:
`[ExcludeFromPaginationNamingConventionTest]` and `[ExcludeFromPaginationApiOptionsConventionTest]`